### PR TITLE
Exchanges

### PR DESCRIPTION
**Pull request by alternativshik**
Adding support for declare and bind/unbind exchanges for Producer

### CODE REVIEW

**Comment by hellysmile** [comment]
@alternativshik can You try to fix lint errors on Python 3.5?

File: aioamqp_consumer/producer.py [hunk 1]
```diff
@@ -6,11 +6,11 @@
 class Producer(AMQPMixin):
 
     def __init__(
-        self,
-        amqp_url,
-        *,
-        amqp_kwargs=None,
-        loop=None
+            self,
```
Comment (hellysmile) [inline aioamqp_consumer/producer.py]: why here is one more ident?

File: aioamqp_consumer/producer.py [hunk 2]
```diff
@@ -126,8 +206,20 @@ def __init__(
...
+            await self._connect()
+
+            await self._exchange_delete(exchange_name, **kwargs)
+        except:  # noqa
+            await self._disconnect()
+            raise
+
     async def _disconnect(self):
         self._known_queues = set()
+        self._known_exchanges = set()
```
Comment (hellysmile) [inline aioamqp_consumer/producer.py]: maybe here we need to reset `_binded_queues `?

**Comment by alternativshik** [comment]
Yes, I'll try.

**Comment by alternativshik** [comment]
Looks OK now for both py3.5 && py3.6

File: aioamqp_consumer/producer.py (hunk 1 above)
Comment (alternativshik) [reply aioamqp_consumer/producer.py]: Oh, PyCharm's fault... Didn't see this ugly autoformat issue.

File: aioamqp_consumer/producer.py (hunk 2 above)
Comment (alternativshik) [reply aioamqp_consumer/producer.py]: Yes, you'r right.

Comment (alternativshik) [reply aioamqp_consumer/producer.py]: May be we have to delete queue from  self._known_queues on queue_delete too?

Comment (hellysmile) [reply aioamqp_consumer/producer.py]: makes sense

File: aioamqp_consumer/producer.py [hunk 3]
```diff
@@ -191,6 +191,9 @@ def __init__(
             await self._connect()
 
             await self._queue_unbind(queue_name, exchange_name, routing_key, **kwargs)
+
+            if queue_name in self._binded_queues:
```
Comment (hellysmile) [inline aioamqp_consumer/producer.py]: just set.discard please

File: aioamqp_consumer/producer.py [hunk 4]
```diff
@@ -202,6 +205,13 @@ def __init__(
             await self._connect()
 
             await self._queue_delete(queue_name, **kwargs)
+
+            if queue_name in self._known_queues:
```
Comment (hellysmile) [inline aioamqp_consumer/producer.py]: just set.discard please

File: aioamqp_consumer/producer.py [hunk 5]
```diff
@@ -202,6 +205,13 @@ def __init__(
...
+                self._known_queues.remove(queue_name)
+
+            # Remove also from binded
+            if queue_name in self._binded_queues:
```
Comment (hellysmile) [inline aioamqp_consumer/producer.py]: just set.discard please

**Comment by hellysmile** [comment]
@alternativshik looks good! I need some to test it more manually. Give me please a couple of days.

There is a couple of things that I am not sure: 

It seems `_exchange_bind` and and `_exchange_unbind` is not used at all.

As well maybe during `queue_unbind` and `exchange_delete` lock maybe need to be used

**Comment by alternativshik** [comment]
Ok.
May be exchange_bind and unbind should be implemented in Producer for better control exchanges and wrap aioamqp lib, but I'm not sure about that. But, I think, this is not a really big problem to implement these methods.

I think you'r right about locks and we should use locks at this point too.

**Comment by hellysmile** [comment]
@alternativshik  Hello there, sorry for such long delay. It will be merged later after some minor changes.

**Comment by alternativshik** [comment]
Hi! Great news!

**Pull request merged**